### PR TITLE
refactor test files

### PR DESCRIPTION
**Pull request by songhappy**
## Description
reorganize test files of Orca Xshards by functionalities

### CODE REVIEW

**Comment by jenniew** [comment]
You can resolve conflict first

File: python/orca/test/bigdl/orca/data/test_pandas_backend.py [hunk 1]
```diff
@@ -122,69 +80,6 @@ def test_read_csv_with_args(self):
...
-        def to_number(d):
-            return 4
-        data_shard = data_shard.transform_shard(to_number)
-        assert len(data_shard) == 2
-        with self.assertRaises(Exception) as context:
-            len(data_shard['abc'])
-        self.assertTrue('No selection operation available for this XShards' in
-                        str(context.exception))
-
     def test_save(self):
```
[the remark this answers is not in the document]
Comment (songhappy) [reply python/orca/test/bigdl/orca/data/test_pandas_backend.py:83]: originally, it was pandas backend, so keep it as it is

File: python/orca/test/bigdl/orca/data/test_pandas_backend.py [hunk 2]
```diff
@@ -195,45 +90,6 @@ def test_save(self):
...
-        col_name = "value"
-        minus_val = 2
-        minus_val_shared_value = SharedValue(minus_val)
-        trans_shard = data_shard.transform_shard(negative, col_name,
-                                                 minus_val_shared_value)
-        data2 = trans_shard.collect()
-        assert data2[0]["value"].values[0] < 0, "value should be negative"
-        assert data[0]["value"].values[0] + data2[0]["value"].values[0] == -2, "value should be -2"
-
     def test_get_item(self):
```
[the remark this answers is not in the document]
Comment (songhappy) [reply python/orca/test/bigdl/orca/data/test_pandas_backend.py:93]: it is panda back with

**Comment by jenniew** [comment]
LGTM

**Pull request merged**